### PR TITLE
Assigned facility: Patient line list

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** https://app.clubhouse.io/simpledotorg/story/776

## Because

We are making changes to our reports to incorporate assigned facility.

## This addresses

This changes the patient line list downloads to be assignd facility compliant. [List of changes](https://docs.google.com/document/d/1YZqm0t5NhTDUkn3Ww8yRqAszKCtAiOqChpFED6U-QXE/edit#heading=h.2v2m3atk9lkd), TLDR:

- Patient line list of a facility now shows assigned patients (before: registered patients) at that facility. 
- Displays assigned facility information in the line list along with registration facility data.

image

### CODE REVIEW

File: app/models/organization_district.rb [hunk 1]
```diff
@@ -13,6 +13,10 @@ def registered_patients
     Patient.where(registration_facility: facilities)
   end
 
+  def assigned_patients
+    Patient.where(assigned_facility: facilities)
+  end
```
Comment (harimohanraj89) [inline app/models/organization_district.rb:18]: Let's make sure this is tested.

File: app/models/facility.rb [hunk 2]
```diff
@@ -36,6 +36,9 @@ class Facility < ApplicationRecord
     -> { with_hypertension },
     class_name: "Patient",
     foreign_key: "registration_facility_id"
+  has_many :assigned_patients,
+    class_name: "Patient",
+    foreign_key: "assigned_facility_id"
```
Comment (harimohanraj89) [inline app/models/facility.rb:41]: Let's add a shouldamatcher test for this new association.

File: spec/jobs/patient_list_download_job_spec.rb [hunk 3]
```diff
@@ -24,4 +24,32 @@
...
+  context "facilities" do
+    let!(:other_facility) { create(:facility) }
+    let!(:assigned_patients) { create_list(:patient, 2, assigned_facility: facility) }
+    let!(:registered_patients) { create_list(:patient, 2, registration_facility: facility, assigned_facility: other_facility) }
+
+    it "should export only assigned patients" do
+      expect(PatientsExporter).to receive(:csv).with(Patient.where(id: assigned_patients))
+      PatientListDownloadJob.perform_now(admin.email, "facility", {facility_id: facility.id}, with_medication_history: false)
+    end
+  end
```
Comment (harimohanraj89) [inline spec/jobs/patient_list_download_job_spec.rb:54]: ❤️

**Comment by prabhanshuguptagit** [comment]
> Claudio/Tanushree/Daniel are on board with adding assigned facility columns to the linelist

Yes, we went over this change with Daniel. 👍

**Pull request merged**